src/gui/torrentcontentmodel.cpp

```diff
@@ -551,20 +564,47 @@ void TorrentContentModel::setupModelData(const BitTorrent::AbstractFileStorage &
     emit layoutChanged();
 }
 
-void TorrentContentModel::selectAll()
+void TorrentContentModel::notifyModelUpdate(const QModelIndex &index)
 {
-    for (int i = 0; i < m_rootItem->childCount(); ++i)
+    Q_ASSERT(index.isValid());
+
+    const int lastColumnIndex = columnCount(index) - 1;
+
+    // emit itself
+    emit dataChanged(index.siblingAtColumn(0), index.siblingAtColumn(lastColumnIndex));
+
+    // propagate up the model
+    QModelIndex parentIndex = parent(index);
+    while (parentIndex.isValid())
     {
-        TorrentContentModelItem* child = m_rootItem->child(i);
-        if (child->priority() == BitTorrent::DownloadPriority::Ignored)
-            child->setPriority(BitTorrent::DownloadPriority::Normal);
+        emit dataChanged(parentIndex.siblingAtColumn(0), parentIndex.siblingAtColumn(lastColumnIndex));
+        parentIndex = parent(parentIndex);
     }
-    emit dataChanged(index(0, 0), index((rowCount() - 1), (columnCount() - 1)));
-}
 
-void TorrentContentModel::selectNone()
-{
-    for (int i = 0; i < m_rootItem->childCount(); ++i)
-        m_rootItem->child(i)->setPriority(BitTorrent::DownloadPriority::Ignored);
-    emit dataChanged(index(0, 0), index((rowCount() - 1), (columnCount() - 1)));
+    // propagate down the model
+    QVector<QModelIndex> parentIndexes;
```

Comment:
Since you effectively use it as stack wouldn't it be better to use QStack instead?

Reply:
No, originally I used a list (for FIFO) but list is inefficient so vector is used instead.
Actually, the iteration order isn't important so I switch to whatever is most efficient.

Reply:
> but list is inefficient so vector is used instead.

Ah, you're still living in the Qt5 era...

Reply:
>Ah, you're still living in the Qt5 era...

Maybe I wasn't clear. I wasn't talking about specific naming/implementations of Qt but general software concepts.
Also I don't see dropping Qt5 any time soon.

Reply:
> Maybe I wasn't clear. I wasn't talking about specific naming/implementations of Qt but general structure concepts.

Maybe it's me who used to specific implementation, but... Are `vector` and `list` different concepts in a general sense? IMO, the concept usually refers to the interface, while the placement of elements in memory is an implementation detail (after all, it is the placement of elements in memory that `QList` and `QVector` differ, isn't it?).

Reply:
>Are vector and list different concepts in a general sense? 

I think so. `vector` is an implementation of Dynamic Array and `list` usually refers to Linked List. You can look them up on Wikipedia.

>IMO, the concept usually refers to the interface, while the placement of elements in memory is an implementation detail (after all, it is the placement of elements in memory that QList and QVector differ in Qt5, isn't it?).

Not familiar with that view... at least in Data Structures textbook they are taught as different concepts.

Reply:
> `list` usually refers to Linked List.

Linked list is a way of implementation of array-like data structure. As stated in documentation, `std::list` is usually implemented as doubly-linked list. But `QList` isn't.
In fact, it turns out that implementation details are often of great importance in such cases, often even influencing the interface. That's why we have many similar classes.
Okay, let's not get into this discussion.

Reply:
>Linked list is a way of implementation of array-like data structure. 

IMO it looks alike but it is different radically, but whatever...

>Okay, let's not get into this discussion.

ok.